.github/workflows/tests.yml

```
@@ -110,9 +102,53 @@ jobs:
           echo "any-plugins=${ANY_PLUGINS}" >> "$GITHUB_OUTPUT"
 
       - name: Select WordPress version
-        if: steps.changed.outputs.any-plugins != 'true' && matrix.wp != 'none'
+        if: matrix.wp != 'none'
         run: .github/files/select-wordpress-tag.sh
 
+      - name: Composer Install
+        run: |
+          # If we're going to be making WorDBless use WP "nightlies", remove the relevant package from Composer's cache to get the latest version.
+          if [[ "$WP_BRANCH" == 'trunk' && ( "$TEST_SCRIPT" == "test-php" || "$TEST_SCRIPT" == "test-coverage" ) ]]; then
+            echo "::group::Clear composer cache for roots/wordpress"
+            DIR=$(composer config cache-files-dir)
+            rm -rf "$DIR/roots/wordpress" "$DIR/roots/wordpress-no-content"
+            echo "::endgroup::"
+          fi
+
+          echo "::group::Composer"
+          composer install --working-dir=tools/php-test-env
+
+          # Add WorDBless for image-cdn package tests since concurrency can nuke uploads that are in use.
+          if [[ "$TEST_SCRIPT" =~ ^test-(php|coverage)$ ]] && jq -e '.name == "automattic/jetpack-image-cdn"' projects/packages/image-cdn/composer.json >/dev/null; then
+            echo "Adding WordBless for image-cdn tests"
+            composer require --working-dir=projects/packages/image-cdn automattic/wordbless:^0.4.2 --dev
+          fi
+
+          # Add WorDBles for publicize package tests since concurrency can nuke uploads that are in use.
+          if [[ "$TEST_SCRIPT" =~ ^test-(php|coverage)$ ]] && jq -e '.name == "automattic/jetpack-publicize"' projects/packages/publicize/composer.json >/dev/null; then
+            echo "Adding WorDBless for publicize tests"
+            composer require --working-dir=projects/packages/publicize automattic/wordbless:^0.4.2 --dev
+          fi
+
+          # Add WorDBles for videopress package tests since concurrency can nuke uploads that are in use.
+          if [[ "$TEST_SCRIPT" =~ ^test-(php|coverage)$ ]] && jq -e '.name == "automattic/jetpack-videopress"' projects/packages/videopress/composer.json >/dev/null; then
+            echo "Adding WorDBless for videopress tests"
+            composer require --working-dir=projects/packages/videopress automattic/wordbless:^0.4.2 --dev
+          fi
```

We can compress this a bit.

Probably a good idea to read the WorDBless version from tools/php-test-env/composer.json instead of hard-coding it.

Also I'm not clear on the point of the `jq -e '.name == "automattic/jetpack-image-cdn"' projects/packages/image-cdn/composer.json` tests, when wouldn't that pass?

And may as well throw in `--no-update` to save running Composer twice in each dir.
```suggestion
          # Add WorDBless for various package tests that use uploads since concurrency can nuke uploads that are in use.
          # @todo Update WorDBless to be able to use a customized upload dir, do that, and remove this.
          if [[ "$TEST_SCRIPT" =~ ^test-(php|coverage)$ ]]; then
            VER=$( jq -r '.require["automattic/wordbless"]' tools/php-test-env/composer.json )
            for pkg in packages/image-cdn packages/publicize packages/videopress; do
              echo "Adding WordBless for $pkg tests"
              composer require --working-dir="projects/$pkg" "automattic/wordbless:$VER" --dev --no-update
            done
          fi
```